flight/targets/sparky2/fw/pios_board.c

```diff
@@ -696,8 +699,20 @@ void PIOS_Board_Init(void) {
 	case HWSPARKY2_ADCDAC_FSKTELEM:
 #if defined(PIOS_INCLUDE_FSK)
 	{
+		uintptr_t fskdac_id;
+		PIOS_FSKDAC_Init(&fskdac_id);
+
 		uintptr_t fskdac_com_id;
-		PIOS_FSKDAC_Init(&fskdac_com_id, &pios_fskdac_config);
+		uint8_t * tx_buffer = (uint8_t *) PIOS_malloc(PIOS_COM_FSKDAC_BUF_LEN);
+		PIOS_Assert(tx_buffer);
+		if (PIOS_COM_Init(&fskdac_com_id, &pios_fskdac_com_driver, fskdac_id,
+		                  NULL, 0,
+		                  tx_buffer, PIOS_COM_FSKDAC_BUF_LEN))
+			panic(6);
```

Comment:
At some point, we should make enums for the `panic()` so that we can guarantee "sameness" across boards.